eth/beacon/db/chain.py

```diff
@@ -0,0 +1,450 @@
+from abc import ABC, abstractmethod
+import functools
+
+from typing import (
+    Dict,
+    Iterable,
+    Tuple,
+)
+from cytoolz import (
+    first,
+    sliding_window,
+)
+
+import rlp
+from eth_utils import (
+    encode_hex,
+    to_tuple,
+    ValidationError,
+)
+from eth_typing import (
+    Hash32,
+)
+
+from eth.db.backends.base import (
+    BaseAtomicDB,
+    BaseDB,
+)
+from eth.constants import (
+    GENESIS_PARENT_HASH,
+)
+from eth.exceptions import (
+    BlockNotFound,
+    CanonicalHeadNotFound,
+    ParentNotFound,
+)
+from eth.validation import (
+    validate_block_number,
+    validate_word,
+)
+
+from eth.beacon.db.schema import SchemaV1
+
+from eth.beacon.types.block import BaseBeaconBlock  # noqa: F401
+
+
+class BaseBeaconChainDB(ABC):
+    db = None  # type: BaseAtomicDB
+
+    # FIXME: Define more abstractmethod
+
+    @abstractmethod
+    def __init__(self, db: BaseAtomicDB) -> None:
+        raise NotImplementedError("BeaconChainDB classes must implement this method")
```

Comment:
We've shifted to just putting `pass` statements here since it's actually impossible to instantiate an `ABC` class without overriding this method, thus it's impossible to actually hit this `raise` statement.

Reply:
My fault, according to `BaseHeaderDB(ABC)`, it might be:
```python
    @abstractmethod
    def __init__(self, db: BaseAtomicDB) -> None:
        self.db = db
```